tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+			b.dist = distB;
+			return polarAngleB - polarAngleA;
+		});
+
+		sortedPoints = sortedPoints.filter((qubit) => {
+			// If the qubit is the lowestYQubit, then keep it
+			if (qubit === lowestYQubit) return qubit;
+			// Get the qubits that have the same polar angle
+			const samepQubits = sortedPoints.filter((q) => q.pAngle === qubit.pAngle);
+			// Get the maximum distance qubit
+			const maxDistQubit = samepQubits.sort((a, b) => {
+				return b.dist - a.dist;
+			})[0];
+			// Return the qubit with the maximum distance
+			return qubit === maxDistQubit;
+		});
+		// Create the stack of qubits which contains the convex hull
+		for (const qubit of sortedPoints) {
+			// Remove qubits from the stack if the angle formed by points next-to-top, top, and qubit is not counterclockwise
+			while (
+				qubitStack.length > 1 &&
+				this._ccw(
+					qubitStack.at(-2), // second from top
+					qubitStack.at(-1), // top of stack
+					qubit
+				) <= 0
+			) {
+				qubitStack.pop();
+			}
+			// push the qubit onto the stack
+			qubitStack.push(qubit);
+		}
+		// Map the qubit stack to the global coordinates
+		const qubitPos = qubitStack.map((qubit) => ({
+			x: qubit.globalX,
+			y: qubit.globalY,
+		}));
+
+		this.beginFill(this.color);
+		// Fill the convex hull
+		this.drawPolygon(qubitPos);
+		this.cursor = 'pointer';
+		this.endFill();
+		// Assign the qubit stack to the plaquette
+		this.qubitStack = qubitStack;
+	};
+
+	_ccw = (p, q, r) => {
+		// Check if the points are counterclockwise
+		const v1 = { x: q.globalX - p.globalX, y: q.globalY - p.globalY };
+		const v2 = { x: r.globalX - q.globalX, y: r.globalY - q.globalY };
+		// Take the cross product
+		const val = -(v1.x * v2.y - v1.y * v2.x); // If val > 0, then counterclockwise, else clockwise
+		if (val === 0) return 0; // Collinear
+		return val; // Clockwise  < 0 or counterclockwise >0
+	};
+
+	createPlaquette = () => {
+		// The graphic should connect the points from the qubits
+		const nQubits = this.qubits.length;
+		if (nQubits < 3) {
+			console.log('Plaquette must have at least 3 qubits');
+			this.plaquetteMade = false;
+			return null;
+		}
+
+		// Create a convex hull
+		this._createConvexHull();
+		const { x, y } = this.calculatePlaquetteCenter();
+		// Set the position and pivot of the plaquette
+		this.pivot.set(x, y);
+		this.position.set(x, y);
+		// Allow for the object to be interactive
+		this.plaquetteMade = true;
+		this.interactive = true;
+		this.cursor = 'pointer';
+		this.makeExtensible();
+		this.toggleCtrlButtons();
+	};
+
+	makeExtensible = () => {
+		// Make the plaquette extensible
+		this.interactive = true;
+		this.buttonMode = true;
+		this.on('pointerdown', this.onDragStart);
+		this.on('pointermove', this.onDragMove);
+		this.on('pointerup', this.onDragEnd);
+	};
+	onDragStart(event) {
+		this.isDragging = true;
+		this.initialPosition = event.data.getLocalPosition(this.parent);
+		console.log(this.initialPosition);
+	}
+
+	onDragMove = (event) => {
```

Comment:
Is this supposed to copy one plaquette at a time?

Reply:
yes

Reply:
I have been able to drag a single plaquette and copy two qubits. I'll play around with it more, maybe capture a video if possible.

Reply:
https://clipchamp.com/watch/NNNNwXJl3lF

This video shows what I mean. Is that intentional?

Reply:
I've also noticed that when I click and drag to create a new tile, a new tile will pop up elsewhere on the tileset.

Reply:
yea that's a bug that needs to be fixed

Reply:
I'll go ahead and create an issue in this repo and mark this convo resolved